kaltura-common/src/rxjs/operators/cancel-on-destroy.ts

```diff
@@ -0,0 +1,50 @@
+import { Observable } from 'rxjs/Observable';
+import { Subject } from 'rxjs/Subject';
+import { OnDestroy } from '@angular/core';
+import 'rxjs/add/observable/merge';
+
+interface EnhancedOnDestroy extends OnDestroy {
+    __ngOnDestroySource__: Subject<string>;
+    __ngOnDestroy__: () => void;
+}
+
+export function cancelOnDestroy<T>(instance: OnDestroy, manualDestroy?: Observable<any>): (source: Observable<T>) => Observable<T> {
+    return (source: Observable<T>) => {
+        if (instance.ngOnDestroy) {
```

Comment:
not really a review comment, more of an advise, feel free to ignore:
I usually prefer to use a more defensive programming approach using guard clauses:
https://refactoring.com/catalog/replaceNestedConditionalWithGuardClauses.html

can remove one level of indentation by replacing the ```if``` logic like so:

```
if (!instance.ngOnDestroy) {
   return source;
}
.... rest of code here, no need else after a return.
```

Reply:
okay, I don't mind